tests/phase1/createunbondingtx.test.ts

```diff
@@ -0,0 +1,176 @@
+    );
+    const invalidStakingTx2 = {
+      ...phase1StakingTransaction,
+      timelock: invalidStakingTxTimelock,
+    }
+    const invalidDelegation2 = {
+      ...phase1Delegation,
+      stakingTx: invalidStakingTx2,
+    }
+    expect(() => staking.createUnbondingTransaction(params, invalidDelegation2))
+      .toThrow("Staking transaction timelock is out of range");
+
+    // Staker public key does not match
+    const invalidStakerPk = dataGenerator.generateRandomKeyPair().publicKeyNoCoord;
+    const invalidDelegation3 = {
+      ...phase1Delegation,
+      stakerPkHex: invalidStakerPk,
+    }
+    expect(() => staking.createUnbondingTransaction(params, invalidDelegation3))
+      .toThrow("Staker public key does not match between connected staker and delegation staker");
+
+    // Invalid staking transaction hex
+    const invalidStakingTxHex = "invalid";
+    const invalidStakingTx3 = {
+      ...phase1StakingTransaction,
+      txHex: invalidStakingTxHex,
+    }
+    const invalidDelegation4 = {
+      ...phase1Delegation,
+      stakingTx: invalidStakingTx3,
+    }
+    expect(() => staking.createUnbondingTransaction(params, invalidDelegation4))
+      .toThrow("Invalid staking transaction hex");
+
+      // Transaction output index is out of range
+    const invalidStakingTx4 = {
+      ...phase1StakingTransaction,
+      outputIndex: dataGenerator.getRandomIntegerBetween(100, 1000),
+    }
+    const invalidDelegation5 = {
+      ...phase1Delegation,
+      stakingTx: invalidStakingTx4,
+    }
+    expect(() => staking.createUnbondingTransaction(params, invalidDelegation5))
+      .toThrow("Staking transaction output index is out of range");
+
+    // StakingTxHashHex does not match from the staking transaction
+    const anotherTx = dataGenerator.generateRandomStakingTransaction(dataGenerator.generateRandomKeyPair())
+    const invalidStakingTxHashHex = anotherTx.stakingTx.getHash().toString("hex");
+    const invalidDelegation6 = {
+      ...phase1Delegation,
+      stakingTxHashHex: invalidStakingTxHashHex,
+    }
+    expect(() => staking.createUnbondingTransaction(params, invalidDelegation6))
+      .toThrow("Staking transaction hash does not match between the btc transaction and the provided staking hash");
+  });
+
+
+  it(`${networkName} should throw an error if fail to build scripts`, async () => {
+    jest.spyOn(require("../../src/staking/stakingScript"), "StakingScriptData").mockImplementation(() => {
+      throw new StakingError(StakingErrorCode.SCRIPT_FAILURE, "build script error");
+    });
+    const phase1Staking = new Phase1Staking(network, stakerInfo);
+
+    expect(() => phase1Staking.createUnbondingTransaction(
+      params,
+      phase1Delegation,
+    )).toThrow("build script error");
+  });
+
+  it(`${networkName} should throw an error if fail to build unbonding tx`, async () => {
+    jest.spyOn(require("../../src/staking"), "unbondingTransaction").mockImplementation(() => {
```

Comment:
Why do we use `require` here instead of regular `import` at the top of the file and `jest.mock()`

https://jestjs.io/docs/mock-functions#mocking-modules